libraries/rib-base/src/main/java/com/badoo/ribs/clienthelper/interactor/Interactor.kt

```diff
@@ -9,17 +12,17 @@ import com.badoo.ribs.core.plugin.NodeLifecycleAware
 import com.badoo.ribs.core.plugin.RibAware
 import com.badoo.ribs.core.plugin.RibAwareImpl
 import com.badoo.ribs.core.plugin.SavesInstanceState
-import com.badoo.ribs.core.plugin.SubtreeChangeAware
 import com.badoo.ribs.core.plugin.ViewAware
 import com.badoo.ribs.core.view.RibView
 
 abstract class Interactor<R : Rib, V : RibView>(
     private val buildParams: BuildParams<*>,
-    private val ribAware: RibAware<R> = RibAwareImpl()
+    private val ribAware: RibAware<R> = RibAwareImpl(),
+    private val childAwareRegistry: ChildAwareRegistry = ChildAwareRegistryImpl(),
 ) : RibAware<R> by ribAware,
     ViewAware<V>,
     NodeLifecycleAware,
-    SubtreeChangeAware,
+    ChildAwareRegistry by childAwareRegistry,
```

Comment:
Blocks descendants of this class from delegating `SubtreeChangeAware` and `NodeAware` via Kotlin.